operator-framework/src/test/java/io/javaoperatorsdk/operator/config/runtime/AnnotationControllerConfigurationTest.java

```diff
@@ -342,6 +350,16 @@ public UpdateControl<ConfigMap> reconcile(ConfigMap resource, Context<ConfigMap>
     }
   }
 
+  @ControllerConfiguration(retry = NoRetry.class)
```

Comment:
@csviri Tried something with Optional but as expected, configuration is a bit weird. We need a marker class because the given class is not a factory but the actual strategy. Then, there is something wrong because nothing prevent from adding `@GradualRetry` with `NoRetry.class`.

Does something like 
```
@ControllerConfiguration(retry = @GradualRetry(maxAttempts = 0))

@ControllerConfiguration(retry = @NoRetry)
````

with the annotation being a factory returning an Optional<Retry> based on the given configuration could be better ?
I think this has been discussed already when introducing GradualRetry annotation in https://github.com/java-operator-sdk/java-operator-sdk/pull/1317 and I think we hit some limitations.

Maybe we can keep the annotation out of controllerConfiguration but need a way to discover annotations that would configure the controllerConfiguration and let them override default configuration.

```
@GradualRetry(maxAttempts = 0)
@ControllerConfiguration() // retry is no longer exposed

@NoRetry
@ControllerConfiguration() // retry is no longer exposed
````